File: examples/raw-python/hmac_request_refund.py
from time import time
from payxcommerce import DEFAULT_BASE_URL, endpoint, hmac_headers, json_body, json_request, print_response

PUBLIC_KEY = "YOUR_PUBLIC_KEY"
SECRET_KEY = "YOUR_SECRET_KEY"

payload = {"transaction_reference": "PXTRX-YYYYMMDD-XXXXXX", "amount": 25.00, "reason": "Customer requested partial refund"}
headers = hmac_headers(PUBLIC_KEY, SECRET_KEY, json_body(payload), f"raw-python-refund-{int(time())}")
print_response(json_request("POST", endpoint(DEFAULT_BASE_URL, "/refunds"), headers, payload))

File: examples/raw-nodejs/hmac-request-refund.js
'use strict';

const { DEFAULT_BASE_URL, endpoint, hmacHeaders, jsonBody, jsonRequest, printResponse } = require('./payxcommerce');

const publicKey = 'YOUR_PUBLIC_KEY';
const secretKey = 'YOUR_SECRET_KEY';
const payload = { transaction_reference: 'PXTRX-YYYYMMDD-XXXXXX', amount: 25.00, reason: 'Customer requested partial refund' };
const headers = hmacHeaders(publicKey, secretKey, jsonBody(payload), `raw-nodejs-refund-${Math.floor(Date.now() / 1000)}`);

jsonRequest('POST', endpoint(DEFAULT_BASE_URL, '/refunds'), headers, payload).then(printResponse).catch((error) => {
  console.error(error.message);
  process.exit(1);
});
